packages/ffe-account-selector-react/src/components/account-selector/AccountSelector.js

```diff
@@ -1,152 +1,189 @@
-import { createAccountFilter } from '../../filter/filters';
-
-class AccountSelector extends Component {
-    constructor(props) {
-        super(props);
-        autoBind(this);
+import { formatIncompleteAccountNumber } from '../../util/format';
 
-        this.baseSelector = null;
+const AccountSelector = ({
+    className,
+    id,
+    locale,
+    selectedAccount,
+    showBalance = false,
+    noMatches,
+    accounts,
+    onAccountSelected,
+    allowCustomAccount = false,
+    labelId,
+    listElementBody,
+    onReset,
+    inputProps,
+    formatAccountNumber = true,
+    ariaInvalid,
+    saveSpaceForDetails = true,
+}) => {
+    const [inputValue, setInputValue] = useState('');
 
-        this.enableFilter = false;
-    }
-
-    renderSuggestion(account) {
-        return (
-            <AccountSuggestion
-                account={account}
-                locale={this.props.locale}
-                showBalance={this.props.showBalance}
-            />
-        );
-    }
+    /*
+     * This matcher function closely resembles the default one of SearchableDropdown,
+     * but it ignores all spaces and periods so that account number formatting won't mess with the search.
+     */
+    const searchMatcher = (searchString, searchAttributes) => item => {
+        const cleanString = value =>
+            `${value}`
+                .replace(/\s/g, '') // Remove all spaces
+                .replace(/\./g, '') // Remove all periods
+                .toLowerCase();
+        const cleanedSearchString = cleanString(searchString);
+        return searchAttributes
+            .map(searchAttribute => cleanString(item[searchAttribute]))
+            .some(cleanedItemAttribute =>
+                cleanedItemAttribute.includes(cleanedSearchString),
+            );
+    };
 
-    assignBaseSelectorRef(baseSelector) {
-        this.baseSelector = baseSelector;
-    }
+    const onInputChange = event => {
+        setInputValue(event.target.value);
+        if (inputProps && inputProps.onChange) {
+            inputProps.onChange();
+        }
+    };
 
-    renderNoMatches() {
-        return (
-            <AccountNoMatch
-                value={this.props.noMatches}
-                locale={this.props.locale}
-            />
-        );
-    }
+    const handleAccountSelected = value => {
+        const hasResetSelection = value === null;
+        const hasSelectedCustomAccount = !value?.accountNumber;
+        if (hasResetSelection) {
+            setInputValue('');
+            onReset();
+        } else if (hasSelectedCustomAccount) {
+            onAccountSelected({ name: '', accountNumber: value.name });
+        } else {
+            onAccountSelected(value);
+        }
+    };
 
-    onAccountSelect(account) {
-        this.enableFilter = false;
-        this.props.onAccountSelected(account);
-    }
+    const formatter = formatAccountNumber
+        ? formatIncompleteAccountNumber
+        : undefined;
```

Comment:
Min tolkning av specen er att dette burde vare null og ikke undefiend. Jag som utvikkler setter ikke ting til og vare undefined. Ting er undefined fordi jag ikke satt dom til noe. 

![image](https://user-images.githubusercontent.com/2248579/117022245-5ef65180-acf8-11eb-990f-d57871000ae0.png)

Reply:
Jeg er forsåvidt enig, men denne blir sendt videre til `SearchableDropdown` hvor `formatter` er en optional funksjon. Så den må bli sendt inn som `undefined` dit hvis vi ikke vil inkludere `null` i typinga i `SearchableDropdown`. Vi kunne evt. ha gjort en sjekk når vi sender den ned som prop til `SearchableDropdown` med `formatter ? formatter : undefined`, men blir ikke det bare to steg istedenfor ett?

Reply:
Det er en mulighet. Før og undgå let pleijer jag gjøre sånn. Funker ju lika bra med let foretrekker personlig const der det er mulig


![image](https://user-images.githubusercontent.com/2248579/117338967-7c195480-ae9f-11eb-94d8-04b7fbcc0cd3.png)